File: 수학(구현)/백준_13015_별찍기23/별찍기23.java
import java.io.*;
import java.util.*;

public class Main{
    public static char[][] arr = new char[200][400];

    public static void set(int n, int r, int idx){
        int row = 2*n-1;
        int col = 4*n-3;

        for(int i = 0; i< col - idx + 1; i++) arr[r][i] = ' ';
a
        if(r == 0 || r == row -1){
            for(int i = 0; i< n; i++) arr[r][i] = '*';
            for(int i = col - n; i< col; i++) arr[r][i] = '*';
        }
        else{
            arr[r][idx] = '*';
            arr[r][idx + n - 1] = '*';
            arr[r][col - idx - 1] = '*';
            arr[r][col - idx - n] = '*';
        }
        for(int i = 0; i< col - idx + 1; i++) System.out.print(arr[r][i]);

    }

    public static void main(String[] args)throws IOException {
        Scanner s = new Scanner(System.in);
        int n = s.nextInt();
        int idx = 0;
        for(int i = 0; i < 2*n-1; i++){
            set(n,i, idx);
            if(i < n-1) idx++;
            else idx--;
            System.out.println();
        }
    }
}

File: 수학(구현)/백준_13015_별찍기23/별찍기23.kt
import java.util.*

var arr = Array(200, {CharArray(400, {' '})})

fun set(n : Int, r: Int,  idx : Int){
    var row = 2*n-1
    var col = 4*n-3

    for(i in 0..col - idx) arr[r][i] = ' '

    if(r == 0 || r == row - 1){
        for(i in 0 until n) arr[r][i] = '*'
        for(i in col - n until col) arr[r][i] = '*'
    }
    else{
        arr[r][idx] = '*'
        arr[r][idx + n - 1] = '*'
        arr[r][col - idx - 1] = '*'
        arr[r][col - idx - n] = '*'
    }
    for(i in 0..col-idx) print(arr[r][i])
}

fun main(args: Array<String>) {
    val s = Scanner(System.`in`)
    var n = s.nextInt()
    var idx = 0
    for(i in 0 until 2*n-1){
        set(n, i, idx)
        if(i < n-1) idx++
        else idx--
        println()
    }
}